src/data/set/basic.lean

```diff
@@ -306,6 +306,9 @@ lemma nonempty_def : s.nonempty ↔ ∃ x, x ∈ s := iff.rfl
 
 lemma nonempty_of_mem {x} (h : x ∈ s) : s.nonempty := ⟨x, h⟩
 
+lemma nonempty_congr (h : ∀ x, x ∈ s ↔ x ∈ t) : s.nonempty ↔ t.nonempty :=
```

Comment:
This lemma looks a bit weird to me - `h` here is just saying that `s = t`. Do you need it for something in this PR?

Reply:
It's for
```
@[simp] theorem finite.nonempty_to_finset {s : set α} (h : s.finite) :
  h.to_finset.nonempty ↔ s.nonempty :=
nonempty_congr (λ _, h.mem_to_finset)
```
It used to use `exists_congr`, which I thought relied on the definitions a little too much, but maybe this relies on the definition of `finset.nonempty` being related to `set.nonempty` a little too much as well.

Reply:
It's for
```
@[simp] theorem finite.nonempty_to_finset {s : set α} (h : s.finite) :
  h.to_finset.nonempty ↔ s.nonempty :=
nonempty_congr (λ _, h.mem_to_finset)
```
It used to use `exists_congr`, which I thought relied on the definitions a little too much, but maybe this relies on the definition of `finset.nonempty` being related to `set.nonempty` a little too much as well.

Reply:
I think I'll remove it and use this proof:
```lean
@[simp] theorem finite.nonempty_to_finset {s : set α} (h : s.finite) :
  h.to_finset.nonempty ↔ s.nonempty :=
by rw [← finset.coe_nonempty, finite.coe_to_finset]
```